File: app/src/main/java/com/pluu/webtoon/di/RealmHelperComponent.java
package com.pluu.webtoon.di;

import com.pluu.webtoon.ui.DetailActivity;
import com.pluu.webtoon.ui.EpisodeFragment;
import com.pluu.webtoon.ui.EpisodesActivity;
import com.pluu.webtoon.ui.IntroActivity;
import com.pluu.webtoon.ui.WebtoonListFragment;

import javax.inject.Singleton;

import dagger.Component;

/**
 * RealmHelperComponent
 * Created by PLUUSYSTEM-SURFACE on 2016-07-12.
 */
@Singleton
@Component(modules = {RealmHelperModule.class})
public interface RealmHelperComponent {

    void inject(IntroActivity activity);
    void inject(EpisodeFragment fragment);
    void inject(WebtoonListFragment fragment);
    void inject(EpisodesActivity activity);
    void inject(DetailActivity activity);

}

File: app/src/main/java/com/pluu/webtoon/di/RealmHelperComponent.kt
package com.pluu.webtoon.di

import com.pluu.webtoon.ui.DetailActivity
import com.pluu.webtoon.ui.EpisodeFragment
import com.pluu.webtoon.ui.EpisodesActivity
import com.pluu.webtoon.ui.WebtoonListFragment
import dagger.Component
import javax.inject.Singleton

/**
 * RealmHelperComponent
 * Created by pluu on 2017-04-30.
 */
@Singleton
@Component(modules = arrayOf(RealmHelperModule::class))
interface RealmHelperComponent {

    fun inject(fragment: WebtoonListFragment)
    fun inject(activity: EpisodesActivity)
    fun inject(fragment: EpisodeFragment)
    fun inject(activity: DetailActivity)

}
